src/General-Rules-Tests/ReSpecShouldNotReferenceToolsRuleTest.class.st

```diff
@@ -0,0 +1,45 @@
+Class {
+	#name : 'ReSpecShouldNotReferenceToolsRuleTest',
+	#superclass : 'ReAbstractRuleTestCase',
+	#instVars : [
+		'classFactory'
+	],
+	#category : 'General-Rules-Tests-Optimization',
+	#package : 'General-Rules-Tests',
+	#tag : 'Optimization'
+}
+
+{ #category : 'running' }
+ReSpecShouldNotReferenceToolsRuleTest >> setUp [
+
+	super setUp.
+	classFactory := ClassFactoryForTestCase new
+]
+
+{ #category : 'running' }
+ReSpecShouldNotReferenceToolsRuleTest >> tearDown [
+
+	classFactory cleanUp.
+	super tearDown
+]
+
+{ #category : 'tests' }
+ReSpecShouldNotReferenceToolsRuleTest >> testRule [
+ 
+	| class |
+	class := classFactory make: [ :builder | builder slots: { } ].
+	class
+		superclass: StPresenter;
+		compile: 'method ^ Smalltalk tools inspector'
+		classified: 'initialization'.
+	self
+		assert: (self myCritiquesOnMethod: class >> #method) size
+		equals: 1 
```

Comment:
```suggestion
	| class |
	class := classFactory make: [ :builder | builder superclass: StPresenter ].
	
	class compile: 'method ^ Smalltalk tools inspector' classified: 'initialization'.
	
	self assert: (self myCritiquesOnMethod: class >> #method) size equals: 1 
```

The superclass should be set directly in the builder block and the slot initialization is useless